### PR TITLE
Merge Login Library changes from WCAndroid

### PR DESCRIPTION
**Pull request by AmandaRiu**
**NOTE** this PR  will not be ready for merge until this [login library PR has been merged](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/37). Use this PR to test the changes, then approve and merge the login library PR. Once that PR is merged this PR will need to pull in the updates from the login library and then be taken out of draft.

This PR merges the small changes to the login library in WCAndroid, this includes:
- [Set the width for the magic link view image assets](https://github.com/woocommerce/woocommerce-android/pull/2266) to `WRAP_CONTENT` so it will scale properly if the image is not a perfect square. 
- [Only set the title in the site address fragment if not a woo login.](https://github.com/woocommerce/woocommerce-android/pull/2330)

Also included are a couple other changes:
* [Gradle version upgraded to 3.5.3](https://github.com/woocommerce/woocommerce-android/pull/1622)
* [Fixed bug when the forgot password url in username password screen was not valid](https://github.com/woocommerce/woocommerce-android/pull/1589)
* [Added logic to fetch the correct SiteModel from the local db for the incoming url](https://github.com/woocommerce/woocommerce-android/pull/1756)

### CODE REVIEW

**Comment by renanferrari** [comment]
I'm assigning this PR for myself, since I'm working with the login library right now for #11785 and I will want to make sure those changes won't conflict with these ones in the end.

**Comment by renanferrari** [comment]
I've reviewed and merged the login library PR. For this one, we just need to update the `enter_site_credentials_instead` string to the same value used in the library, so that the `strings-check` won't fail.

@AmandaRiu I'm not familiar with the usual process here, but I'm assuming I can just push this change to this branch myself? Or this something you should do?

**Comment by AmandaRiu** [comment]
Updated with the latest changes from the login library and took this PR out of draft. 

> @AmandaRiu I'm not familiar with the usual process here, but I'm assuming I can just push this change to this branch myself? Or is this something you should do?

Typically the PR author would make the change and push it to the branch, but I'm not sure what you mean. Can you give me more info?

File: libs/login/WordPressLoginFlow/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <string name="open_mail">Open mail</string>
     <string name="alternatively">Alternatively:</string>
     <string name="enter_site_address_instead">Log in by entering your site address.</string>
-    <string name="enter_site_credentials_instead">Log in with site credentials</string>
+    <string name="enter_site_credentials_instead">Log in with site credentials.</string>
```
Comment (renanferrari) [inline libs/login/WordPressLoginFlow/src/main/res/values/strings.xml:25]: This change should be replicated [here](https://github.com/wordpress-mobile/WordPress-Android/blob/16fdfdad8ded1d2ae9cff3c9d97ed2cc2066ca20/WordPress/src/main/res/values/strings.xml#L2414) for the sake of linting.

Comment (AmandaRiu) [reply libs/login/WordPressLoginFlow/src/main/res/values/strings.xml:25]: Fixed in 66ea2cd38c5e57f33d1ed55579063e15b508b8f0 👍

**Comment by AmandaRiu** [comment]
@renanferrari Resolved the string issue. Ready for another round!

**Pull request merged**